packages/opentelemetry-resources/test/detectors/BrowserDetector.test.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as sinon from 'sinon';
+import { Resource } from '../../src';
+import { browserDetector } from '../../src/platform/browser/detectors';
+import {
+  assertResource,
+  assertEmptyResource,
+} from '../util/resource-assertions';
+
+
+describe('browserDetector()', () => {
+  beforeEach(() => {
+    global.window = {
```

Comment:
`sinon.stub` should be preferred to create mocks. Also `global` is not a standard _global_ alias in browser environment, please use `globalThis` instead.

Reply:
updated.
tried to 
```
sinon.stub(globalThis, 'window')
....
```
but got `TypeError: Cannot stub non-existent property window`

Reply:
My guess is that these tests are run on the Node.js environment, which is not expected. I'm filed https://github.com/open-telemetry/opentelemetry-js/pull/2739 to fix this. I don't want to block this PR for #2739. So I'm ok with this as is.